### PR TITLE
Use new data platform lib in nms-magmalte

### PR DESCRIPTION
**Pull request by ghislainbourgeois**
# Description

Use new data platform library for postgresql integration in nms-magmalte-operator, replacing the deprecated `ops-lib-pgsql`.

## Checklist

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.
- [ ] I have bumped the version of any required library.

### CODE REVIEW

File: nms-magmalte-operator/tests/unit/test_charm.py [hunk 1]
```diff
@@ -133,10 +136,11 @@ def test_given_pod_is_leader_when_database_relation_joined_event_then_database_i
             postgres_host,
             postgres_port,
         )
-        self.harness.charm._on_database_relation_joined(db_event)
+        self.harness.charm._on_magma_nms_magmalte_pebble_ready(db_event)
```
Comment (sanchezfdezjavier) [inline nms-magmalte-operator/tests/unit/test_charm.py]: Please, change the name of the test `test_given_pod_is_leader_when_database_relation_joined_event_then_database_is_set_correctly` -> `test_given_pod_is_leader_when_pebble_ready_event_then_database_is_set_correctly`.
I can't comment in the test signature :(

File: nms-magmalte-operator/tox.ini [hunk 2]
```diff
@@ -12,7 +12,7 @@ integration_test_path = {toxinidir}/tests/integration/
 all_path = {[vars]src_path} {[vars]unit_test_path} {[vars]integration_test_path}
```
Comment (sanchezfdezjavier) [inline nms-magmalte-operator/tox.ini]: IDK if we already want to outsource test dependencies to a separate file, It's done by Yazan in this [PR](https://github.com/canonical/charmed-magma-orchestrator/pull/250/files) and the Juju version is not exactly the same.

**Review by sanchezfdezjavier — APPROVED** [review]
Aside, from the comments LGTM

**Review by Gmerold — APPROVED** [review]

File: nms-magmalte-operator/tests/unit/test_charm.py [hunk 3]
```diff
@@ -40,11 +40,16 @@ class TestCharm(unittest.TestCase):
     )
```
Comment (sanchezfdezjavier) [inline nms-magmalte-operator/tests/unit/test_charm.py]: We also need to update the interface name in the integration tests.

File: nms-magmalte-operator/tox.ini (hunk 2 above)
Comment (ghislainbourgeois) [reply nms-magmalte-operator/tox.ini]: I removed those changes, as they did not fix the integration tests. Let's do this in another PR.

File: nms-magmalte-operator/tests/unit/test_charm.py (hunk 1 above)
Comment (ghislainbourgeois) [reply nms-magmalte-operator/tests/unit/test_charm.py]: I changed it, but to `test_given_pod_is_leader_when_database_created_event_then_database_is_set_correctly`, and also renamed the method handling the events.

File: nms-magmalte-operator/tests/unit/test_charm.py (hunk 3 above)
Comment (ghislainbourgeois) [reply nms-magmalte-operator/tests/unit/test_charm.py]: Good catch, fixed!

File: nms-magmalte-operator/tests/integration/test_integration.py [hunk 4]
```diff
@@ -77,7 +77,7 @@ async def _deploy_orc8r_certifier(self, ops_test):
             series="jammy",
         )
```
Comment (sanchezfdezjavier) [inline nms-magmalte-operator/tests/integration/test_integration.py]: In line 46 we probably want to add the channel
```python
await ops_test.model.deploy(
             "postgresql-k8s",
             application_name=DB_APPLICATION_NAME,
             channel="14/stable",
         )
```

Comment (ghislainbourgeois) [reply nms-magmalte-operator/tests/integration/test_integration.py]: Done

File: nms-magmalte-operator/metadata.yaml [hunk 5]
```diff
@@ -43,7 +43,7 @@ provides:
 
 requires:
   db:
-    interface: pgsql
+    interface: postgresql_client
```
Comment (ghislainbourgeois) [inline nms-magmalte-operator/metadata.yaml:46]: Also needs changes to the bundle for the relation endpoint, needs to be `database`.

**Review by gruyaume — APPROVED** [review]

**Pull request merged**